core/stackref_test.go

```diff
@@ -55,4 +63,90 @@ func TestStackRef(t *testing.T) {
 	if !testutils.EqualInt64Slice(observedVals, expectedVals) {
 		t.Error("iterating through the stackref should return the elements in the order they were pushed. Expected:", expectedVals, ", got: ", observedVals)
 	}
+
+	sr2 := core.NewStackRef()
+	obj1 := core.NewObj(10, -1, core.OBJ_TYPE_STRING, core.OBJ_ENCODING_INT)
+	core.Put("key1", obj1)
+	sr2.Push("key1")
+
+	if key, err := sr2.Pop(); err != nil || key.Obj != core.Get("key1") {
+		t.Error("popping non-expired key from stackref should pop element")
+
+	}
+
+	keys := []struct {
+		key           string
+		value         int
+		expDurationMs int64
+	}{
+		{"key0", 0, 0},     // expired key
+		{"key1", 10, 0},    // expired key
+		{"key2", 20, -1},   // non-expired key
+		{"key3", 30, 0},    // expired key
+		{"key4", 40, 0},    // expired key
+		{"key5", 50, 0},    // expired key
+		{"key6", 60, -1},   // non-expired key
+		{"key7", 70, 0},    // expired key
+		{"key8", 80, -1},   // non-expired key
+		{"key9", 90, -1},   // non-expired key
+		{"key10", 100, -1}, // non-expired key
+	}
+
+	for _, k := range keys {
+		createAndPushKey(sr2, k.key, k.value, k.expDurationMs)
+	}
+
+	expectedVals = []int64{100, 90, 80}
+	observedVals = make([]int64, 0, 3)
+
+	for i := 0; i < len(expectedVals); i++ {
+		key, err := sr2.Pop()
+		if err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+		if key == nil {
+			t.Errorf("expected key, got nil")
+		} else {
+			observedVals = append(observedVals, int64(key.Obj.Value.(int)))
+		}
+	}
+
+	if !testutils.EqualInt64Slice(observedVals, expectedVals) {
+		t.Errorf("expected %v, got %v", expectedVals, observedVals)
+	}
+
+	if key, err := sr2.Pop(); err != nil || int64(key.Obj.Value.(int)) != 60 {
+		t.Errorf("popping an expired key followed by a non-expired key should work")
+	}
+
+	if key, err := sr2.Pop(); err != nil || int64(key.Obj.Value.(int)) != 20 {
+		t.Errorf("popping multiple expired keys followed by a non-expired key should work")
+	}
+
+	if _, err := sr2.Pop(); err != core.ErrStackEmpty {
+		t.Errorf("popping all expired keys till stack becomes empty should work")
+	}
+
+}
+
+func BenchmarkRemoveLargeNumberOfExpiredKeys(b *testing.B) {
+
+	for _, v := range keys {
+		populateData(v.count)
```

Comment:
Why are we populating the data here when the benchmark function is also adding more keys.

Reply:
Yes, you're right. I thought I needed all the configuration, but I will add only the necessary configurations from populateData.